From d76d6158fb60b0538973afd30d20d88420e9ad1f Mon Sep 17 00:00:00 2001
From: "Erich E. Hoover" <erich.e.hoover@gmail.com>
Date: Mon, 7 Jul 2025 15:05:28 -0500
Subject: [PATCH 29/48] kernelbase: Translate FILE_FLAG_OPEN_REPARSE_POINT.

(cherry picked from commit fbd8c97d2ecfa65e14130c24f3ac5cedd720b177)

CW-Bug-Id: #26492
---
 dlls/kernelbase/file.c  | 2 ++
 dlls/ntdll/tests/file.c | 9 +++------
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/dlls/kernelbase/file.c b/dlls/kernelbase/file.c
index 04a615adf9e..b56abbac413 100644
--- a/dlls/kernelbase/file.c
+++ b/dlls/kernelbase/file.c
@@ -761,6 +761,8 @@ static UINT get_nt_file_options( DWORD attributes )
         options |= FILE_DELETE_ON_CLOSE;
     if (attributes & FILE_FLAG_NO_BUFFERING)
         options |= FILE_NO_INTERMEDIATE_BUFFERING;
+    if (attributes & FILE_FLAG_OPEN_REPARSE_POINT)
+        options |= FILE_OPEN_REPARSE_POINT;
     if (!(attributes & FILE_FLAG_OVERLAPPED))
         options |= FILE_SYNCHRONOUS_IO_NONALERT;
     if (attributes & FILE_FLAG_RANDOM_ACCESS)
diff --git a/dlls/ntdll/tests/file.c b/dlls/ntdll/tests/file.c
index 833e578485c..6f19c9de76c 100644
--- a/dlls/ntdll/tests/file.c
+++ b/dlls/ntdll/tests/file.c
@@ -6543,7 +6543,7 @@ static void test_reparse_points(void)
     ret = GetFinalPathNameByHandleW( handle2, ret_path, ARRAY_SIZE( ret_path ), VOLUME_NAME_DOS );
     ok( ret > 0, "got error %lu\n", GetLastError() );
     swprintf( path, ARRAY_SIZE(path), L"\\\\?\\%stestreparse_dirlink", temp_path );
-    todo_wine ok( !wcscmp( ret_path, path ), "expected path %s, got %s\n", debugstr_w( path ), debugstr_w( ret_path ));
+    ok( !wcscmp( ret_path, path ), "expected path %s, got %s\n", debugstr_w( path ), debugstr_w( ret_path ));
     CloseHandle( handle2 );
 
     status = NtFsControlFile( handle, NULL, NULL, NULL, &io, FSCTL_DELETE_REPARSE_POINT, NULL, 0, NULL, 0 );
@@ -6614,11 +6614,8 @@ static void test_reparse_points(void)
     swprintf( path, ARRAY_SIZE(path), L"%s/testreparse_dirlink", temp_path );
     handle2 = CreateFileW( path, DELETE, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, NULL,
                            OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OPEN_REPARSE_POINT, 0 );
-    todo_wine ok( handle2 != INVALID_HANDLE_VALUE, "got error %lu\n", GetLastError() );
-    if (handle2 != INVALID_HANDLE_VALUE)
-        status = NtSetInformationFile( handle2, &io, &fdi, sizeof(fdi), FileDispositionInformation );
-    else
-        status = NtSetInformationFile( handle, &io, &fdi, sizeof(fdi), FileDispositionInformation );
+    ok( handle2 != INVALID_HANDLE_VALUE, "got error %lu\n", GetLastError() );
+    status = NtSetInformationFile( handle2, &io, &fdi, sizeof(fdi), FileDispositionInformation );
     ok( !status, "got %#lx\n", status );
     CloseHandle( handle2 );
 
-- 
2.53.0

